javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -1048,11 +1048,15 @@ public int length() {
 
     @Override
     public Array<T> reverse() {
-        final Object[] arr = new Object[delegate.length];
-        for (int i = 0; i < delegate.length; i++) {
-            arr[delegate.length - 1 - i] = delegate[i];
+        if (size() <= 1) {
+            return this;
+        } else {
+            final Object[] arr = new Object[delegate.length];
+            for (int i = 0; i < delegate.length; i++) {
+                arr[delegate.length - 1 - i] = delegate[i];
+            }
```

Comment:
@AlparSzabados please use the following snippet.

```java
final int length = delegate.length;
for (int i = 0, j = length - 1; i < length; i++, j--) {
    arr[j] = delegate[i];
}
```

(Too much discussion for minor things here)